packages/data/src/world/mq/dodongo_cavern_mq.yml

```diff
@@ -2,159 +2,279 @@
-    "Dodongo Cavern Room Before Upper Lizalfos": "can_use_sticks || has_fire"
-  locations:
-    "MQ Dodongo Cavern Larvae Room Chest": "soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && (can_use_sticks || has_fire)"
-    "MQ Dodongo Cavern GS Larve Room": "gs && (can_use_sticks || has_fire)"
-    "MQ Dodongo Cavern Heart Vanilla Bomb Bag Room": "true"
-"Dodongo Cavern Upper Lizalfos":
+    "Dodongo Cavern Main Room Lower": "true"
+    "Dodongo Cavern Room After Upper Staircase": "true"
+    "Dodongo Cavern Main Room Upper Skull Side": "longshot_anywhere"
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "true"
+"Dodongo Cavern Vanilla Bomb Bag Room Lower":
   dungeon: DC
   exits:
-    "Dodongo Cavern Upper Ledges": "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
+    "Dodongo Cavern Main Room Upper Entrance Side": "true"
+    "Dodongo Cavern Room Before Upper Lizalfos": "can_use_sticks || has_fire"
+    "Dodongo Cavern Vanilla Bomb Bag Room Upper": "can_hookshot || has_hover_boots || (is_adult && trick(OOT_DC_JUMP)) || climb_anywhere"
+    "Dodongo Cavern MQ Larvae Room": "event(OPEN_MQ_DC_LARVAE_ROOM)"
+  events:
+    OPEN_MQ_DC_LARVAE_ROOM: "can_use_sticks || has_fire"
   locations:
-    "MQ Dodongo Cavern GS Upper Lizalfos": "gs && has_explosives_or_hammer"
-    "MQ Dodongo Cavern Pot Miniboss 1": "true"
-    "MQ Dodongo Cavern Pot Miniboss 2": "true"
-    "MQ Dodongo Cavern Pot Miniboss 3": "true"
-    "MQ Dodongo Cavern Pot Miniboss 4": "true"
-"Dodongo Cavern Upper Ledges":
+    "MQ Dodongo Cavern Heart Vanilla Bomb Bag Room": "true"
+    # Fails to load
+    "MQ Dodongo Cavern Grass Vanilla Bomb Bag Room": "can_cut_grass"
+"Dodongo Cavern Vanilla Bomb Bag Room Upper":
   dungeon: DC
   exits:
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "true"
+    "Dodongo Cavern Main Room Upper Skull Side": "event(DC_MQ_SHORTCUT) || climb_anywhere"
     "Dodongo Cavern Room After Upper Lizalfos": "true"
-    "Dodongo Cavern Torch Room": "true"
   events:
-    "DC_MQ_SHORTCUT": "true"
+    DC_MQ_SHORTCUT: "has_bombflowers || has_fire || has_explosives_or_hammer"
   locations:
     "MQ Dodongo Cavern Upper Ledge Chest": "true"
     "MQ Dodongo Cavern Pot Vanilla Bomb Bag Room 1": "true"
     "MQ Dodongo Cavern Pot Vanilla Bomb Bag Room 2": "true"
+"Dodongo Cavern MQ Larvae Room":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "event(MQ_DC_CLEAR_LARVAE_ROOM)"
+  events:
+    MQ_DC_CLEAR_LARVAE_ROOM: "soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && can_damage"
+  locations:
+    "MQ Dodongo Cavern Larvae Room Chest": "event(MQ_DC_CLEAR_LARVAE_ROOM)"
+    "MQ Dodongo Cavern GS Larve Room": "gs && can_damage_skull"
+"Dodongo Cavern Room Before Upper Lizalfos":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Upper Lizalfos": "(can_use_sticks && has_goron_bracelet) || has_explosives_or_hammer || has_blue_fire || has_blue_fire_arrows_mudwall" #Needs macro for getting past enemies still in path
```

Comment:
Remove blue fire. Also, you still need to burn some webs to get to the end.

Reply:
tell me what to put for the webs. i cant picture it

Reply:
ah. no webs, but for the sticks you would need to be able to get the torch lit. So probably an event for the torch on the other side (requires lighting the two torches on the front side).